### ISSUE TITLE
columnOptions nullable default value is false?

### ISSUE THREAD

**Issue by luozhihua** [typeorm/typeorm#2748]
**Issue type:**

[x] question
[ ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [ ] `mysql`
    [x] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
!>

**Comment by gintsgints** [comment]
Yes. It is false by default. Please close Issue if answer helps.

**Comment by IRCraziestTaxi** [comment]
I'd like to expand on this although I cannot reopen the issue.

I am pretty sure that without specifying column options at all, the default behavior is for a column to allow nulls.

That being the case, I expected the same when specifying column options but _not_ specifying a value for `nullable`.

Why the difference in behavior?

**Comment by IRCraziestTaxi** [comment]
Update: I swear when using `@Column()` (with no options at all) the default behavior used to be nullable true... that no longer appears to be the case. :/

**Comment by Blockost** [comment]
> Update: I swear when using `@Column()` (with no options at all) the default behavior used to be nullable true... that no longer appears to be the case. :/

This exact same thing happened to me just now. I had to manually specify `{nullable: false}` on every column but it seems, it's the default behavior right now...

**Comment by fan-tom** [comment]
>Actually, it depends on the decorator used to declare the column. It seems that @Column is non-nullable by default but when you declare a relation with @ManyToOne it is nullable by default.

It surprised me too, that relations are nullable by default whereas ordinary fields are not null by default, what is the logic behind this behaviour?

**Comment by artemkozlenkov** [comment]
it seems like documentation could've been written in more details

**Comment by ORESoftware** [comment]
Can someone just show how to make a column not nullable?

```
  @Column()
  @Nullable(false).   // i made this up  :(
  email: string;
```

**Comment by nebkat** [comment]
https://typeorm.io/#/entities/column-options

You should use the community [Slack](https://join.slack.com/t/typeorm/shared_invite/zt-gej3gc00-hR~L~DqGUJ7qOpGy4SSq3g) for general questions like this.

**Comment by luozhihua** [comment]
@ORESoftware 

```typescript
@Column({ nullable: false })
email: string;
```
